src/TwigComponent/doc/index.rst

```diff
@@ -1046,6 +1046,108 @@ And in your component template you can access your embedded block
         {% block footer %}{% endblock %}
      </div>
 
+Anonymous TwigComponent
+-----------------------
+
+Sometimes a component is simple enough that it doesn't have any complex logic or injected services.
+In this case, you can skip the class and only create the template. The component name is determined
+by the location of the template (see `Twig Template Namespaces`_):
+
+.. code-block:: html+twig
+
+    {# templates/components/Button/Primary.html.twig #}
+    <button class="primary">
+        {% block content %}
+        {% endblock %}
+    </button>
+
+Then use your component with ``:`` to navigate through sub-directories (if there are any):
+
+.. code-block:: html+twig
+
+     {# index.html.twig #}
+     ...
+    <div>
+       <twig:Button:Primary>
+           Click Me!
+       </twig:Button:Primary>
+    </div>
+
+Still get all your attributes by rendering the ``attributes`` object in your template:
+
+.. code-block:: html+twig
+
+    {# templates/components/Button.html.twig #}
+    <button {{ attributes }}>
+        {% block content %}
+        {% endblock %}
+    </button>
+
+Then pass any further data to your reusable "button" like so:
+
+.. code-block:: html+twig
+
+     {# index.html.twig #}
+     ...
+    <div>
+        <twig:Button role="button" class="foo">
+            Click Me!
+        </twig:Button>
+    </div>
+
+    {# renders as: #}
+    <button role="button" class="foo">Click Me!</button>
+
+Now let's pass props to your "button" by defining a ``{% props %}`` tag in its view.
```

Comment:
Let's reverse this section: show the usage first, then show the `{% props` syntax. Something like:

You can also pass a variable (prop) into your template:

<twig:Button role="button" icon="fa-plus" type="primary">Buy now!</twig:Button>

To tell the system that `icon` and `type` are props and not attributes, use the `{% props` tag at the top of your template:

```
{% props icon, type = 'primary' %}

<button {{ attributes.defaults({
    class: 'btn btn-'~type
} }}>
    {% block content %}{% endblock %}
    {% if icon %}
        <span class="fa-solid fa-{{ icon }}"></span>
    {% endif %}
</button>
```

I don't love the `attributes.defaults()` syntax, but it does make the components more flexible.